onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorElementWise.cpp

```diff
@@ -554,7 +554,9 @@ class DmlOperatorElementwiseQLinear : public DmlOperator
             // If the input tensor is a 1D vector, then extra massaging is needed to project their
             // 1D vectors back to the full shape for broadcasting along the given axis.
             // The 1D vector should have a length equal to the output tensor's dimension on that axis.
-            if (inputTensorShape.size() == 1 && inputTensorShape != outputShape)
+            if ((inputTensorShape.size() == 1 ) && 
+                (outputShape.size() > 1) &&
```

Comment:
This line is dubious, as the output shape rank shouldn't matter.

The constraint in simpler terms is that if `scale.rank == 1`, then `scale.shape[0] = input.shape[axis]`.

Reply:
If we want to permit 1D tensors of 1 element too, and treat them like 0D tensors, then we could say:

`ML_CHECK_VALID_ARGUMENT(scale.shape[0] = input.shape[axis] || scale.shape[0] == 1)`